### PR TITLE
Backport: Add sanitize_log_messages and remove queryserver-config-terse-errors impact on log messages

### PR DESCRIPTION
**Pull request by mattlord**
## Description
This backports two related PRs together. Please see https://github.com/vitessio/vitess/pull/9634 for additional details.

This maintains consistent/unchanged behavior for `-queryserver-config-terse-errors` between 12.0 and 13.0 while adding support for sanitizing log messages with the new `-sanitize_log_messages` flag.

## Related Issue(s)
Backports: 
 - https://github.com/vitessio/vitess/pull/9550
 - https://github.com/vitessio/vitess/pull/9634

## Checklist
- [x] Should this PR be backported? NO
- [x] Tests were added
- [x] Documentation PR: https://github.com/vitessio/website/pull/969

### CODE REVIEW

**Review by GuptaManan100 — APPROVED** [review]
We should merge this after #9634 is merged

**Review by deepthi — COMMENTED** [review]
I agree with back porting #9634. What is the justification for back porting #9550?
In general we don't add new features to the release branch once RC1 is out.

**Comment by mattlord** [comment]
> I agree with back porting #9634. What is the justification for back porting #9550?
> In general we don't add new features to the release branch once RC1 is out.

The combination of the two ensures that:
1. We maintain consistent behavior for TerseErrors between 12.0 and 13.0
2. For anyone using TerseErrors in 13.0-SNAPSHOT, we’re not removing behavior they likely wanted (log sanitization)  w/o a (better) replacement 

And given that it’s a minor change (log msg related), it seems like a good exception IMO.

I can pare the backport down to just 1 from above though if you want.

Thanks!

**Review by deepthi — APPROVED** [review]

**Pull request merged**
